pkg/repository/v1/sqlcv1/queue.sql

```diff
@@ -90,6 +92,8 @@ WHERE
     )
     -- Added to ensure that the index is used
     AND qi.priority >= 1 AND qi.priority <= 4
+    -- Filter out those that are older than tenant's retention period
+    AND qi.task_inserted_at >= NOW() - t."dataRetentionPeriod"::interval
```

Comment:
any idea if this will change the plan for this query meaningfully? the task inserted at isn't part of the PK (the other columns in this where clause all are) but it's used in a separate index too